src/cryfs-cli/program_options/ProgramOptions.cpp

```diff
@@ -16,6 +16,20 @@ ProgramOptions::ProgramOptions(const bf::path &baseDir, const bf::path &mountDir
     :_baseDir(baseDir), _mountDir(mountDir), _configFile(configFile), _foreground(foreground),
      _cipher(cipher), _blocksizeBytes(blocksizeBytes), _unmountAfterIdleMinutes(unmountAfterIdleMinutes),
      _logFile(logFile), _fuseOptions(fuseOptions) {
+
+    auto hasNoOption = [&](const char *opt) {
+        for (const string& it : _fuseOptions) {
+            if (std::strncmp(it.c_str(), opt, std::strlen(opt))) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (hasNoOption("subtype=cryfs") && hasNoOption("fsname=cryfs@")) {
+               _fuseOptions.push_back("-ofsname=cryfs@"+baseDir.native());
+               _fuseOptions.push_back("-osubtype=cryfs");
+    }
```

Comment:
I initially added it there but changed my mind because Fuse class does not have "baseDir" entry or i couldnt find it and changing the signature of the class felt like making too many changes.

Two alternatives i considered were:
1. Add additional "boost::filesystem::path" argument that passes in baseDir.
2. Change the second argument and pass in "ProgramOptions" object instead of the current "vector<string>" object.

Which one of these two do you prefer?